CRM/Core/Form/EntityFormTrait.php

```diff
@@ -159,19 +163,26 @@ protected function addFormButtons() {
    */
   protected function getEntityDefaults() {
     $defaults = [];
-    if ($this->_action != CRM_Core_Action::DELETE &&
+    if (!$this->isDeleteContext() &&
       $this->getEntityId()
     ) {
       $params = ['id' => $this->getEntityId()];
       $baoName = $this->_BAOName;
       $baoName::retrieve($params, $defaults);
     }
-    foreach ($this->entityFields as $fieldSpec) {
+    foreach ($this->entityFields as &$fieldSpec) {
       $value = CRM_Utils_Request::retrieveValue($fieldSpec['name'], $this->getValidationTypeForField($fieldSpec['name']));
       if ($value !== FALSE && $value !== NULL) {
         $defaults[$fieldSpec['name']] = $value;
       }
+      if (CRM_Utils_Array::value('formatter', $fieldSpec) == 'crmMoney') {
```

Comment:
I think you have to do this at the end of the function - you are looping through the fields - but it will only work if currency field is earlier in the array than the field-to-be-formatted

Reply:
@mattwire this comment here ^^ is the one that I think matters